### PR TITLE
enhancement(helm platform): Option to add additional ports to the service in vector-agent helm chart

### PR DESCRIPTION
**Pull request by MonolithProjects**
This PR is adding an option to the vector-agent helm chart to add extra ports to the service resource used for [Topology-aware traffic routing](https://kubernetes.io/docs/concepts/services-networking/service-topology/). So now also the traffic from the sources can be routed based on the topology of the cluster.

### CODE REVIEW

File: distribution/helm/vector-agent/values.yaml [hunk 1]
```diff
@@ -30,16 +30,26 @@ fullnameOverride: ""
...
-  #   - "kubernetes.io/hostname"
-  #   - "topology.kubernetes.io/zone"
-  #   - "topology.kubernetes.io/region"
-  #   - "*"
+  type: ClusterIP
+  topologyKeys:
+    - "kubernetes.io/hostname"
+    - "topology.kubernetes.io/zone"
+    - "topology.kubernetes.io/region"
+    - "*"
```
Comment (spencergilbert) [inline distribution/helm/vector-agent/values.yaml:42]: Can we leave these out by default?

File: distribution/helm/vector-agent/values.yaml [hunk 2]
```diff
@@ -30,16 +30,26 @@ fullnameOverride: ""
...
+  ports:
+    - name: api
+      port: 80
+      protocol: TCP
+      targetPort: 80
```
Comment (spencergilbert) [inline distribution/helm/vector-agent/values.yaml:47]: 🤔 I'm curious about moving this to a higher port, there's no need for us to add privileges if we don't need to. I didn't think of this on the PR that added this, perhaps we should default this to 8686 - the [default port for the API](https://vector.dev/docs/reference/api/#address).

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: distribution/helm/vector-agent/values.yaml (hunk 2 above)
Comment (MonolithProjects) [reply distribution/helm/vector-agent/values.yaml:47]: Yeah, totally agree. Will fix this.

File: distribution/helm/vector-agent/values.yaml (hunk 1 above)
Comment (MonolithProjects) [reply distribution/helm/vector-agent/values.yaml:42]: Sure.

**Review by spencergilbert — APPROVED** [review]
Great - thanks!

**Pull request merged**